### PR TITLE
[TAS-1816] 🚸 Display the reset password hint when hiding social options

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2024-07-08 下午6 52 45

### CODE REVIEW

**Comment by nwingt** [comment]
Use i18n?

**Comment by AuroraHuang22** [comment]
@nwingt  updated
[🌐 Apply i18n to hint](https://github.com/likecoin/likecoin-wallet-connector/pull/60/commits/a42c288ff73227e73833fc06ad363dbcfe95b676)

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/components/authcore-dialog.tsx [hunk 1]
```diff
@@ -28,6 +31,16 @@ export const AuthcoreDialog: FC<AuthcoreDialogProps> = ({
       onClose={closeDialog}
     >
       <div id={AUTHCORE_DIALOG_ID} />
+      {isHideSocialLogin && (
+        <div className="lk-flex lk-flex-col lk-flex-col lk-w-full lk-items-center lk-gap-y-2 lk-mt-[12px]">
+          <p className="lk-text-like-gray lk-text-center lk-w-full lk-text-[12px]">
+            <FormattedMessage id="wallet_connect_hint_reset_password"/>
+            <span className='lk-text-like-green'>
+              <FormattedMessage id="wallet_connect_hint_reset_password_highlight"/>
+            </span>
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply src/components/authcore-dialog.tsx]: [🎨Use rich text formatting](https://github.com/likecoin/likecoin-wallet-connector/pull/60/commits/74ee268b3ddefb6212e0e6038ddf3d31a145c09c)

File: src/i18n/translations/en.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
   "wallet_connect_header_title_open_app": "Launch App",
   "wallet_connect_header_title_scan_qrcode": "Scan QR Code",
   "wallet_connect_hint_approve": "Please approve the connection request in the app by clicking the button below.",
+  "wallet_connect_hint_reset_password": "If you can't find your login method,<span>reset your password.</span>",
```
Comment (nwingt) [inline src/i18n/translations/en.json:25]:
```suggestion
  "wallet_connect_hint_reset_password": "If you can't find your login method,<span>reset your password</span>.",
```

File: src/i18n/translations/zh.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
   "wallet_connect_header_title_open_app": "啟動 App",
   "wallet_connect_header_title_scan_qrcode": "掃描二維碼",
   "wallet_connect_hint_approve": "請點擊以下按鈕打開 App 並批准連接請求。",
+  "wallet_connect_hint_reset_password": "若找不到您原先的登入方式，請<span>重設密碼</span>",
```
Comment (nwingt) [inline src/i18n/translations/zh.json:25]:
```suggestion
  "wallet_connect_hint_reset_password": "若找不到您原先的登入方式，請<span>重設密碼</span>。",
```

Comment (AuroraHuang22) [reply src/i18n/translations/zh.json]: [💬 Adjust wording](https://github.com/likecoin/likecoin-wallet-connector/pull/60/commits/5f187b73f617a9e44505fc1f1bfa60ecea3ae5ba)

File: src/i18n/translations/en.json (hunk 2 above)
Comment (AuroraHuang22) [reply src/i18n/translations/en.json]: [💬 Adjust wording](https://github.com/likecoin/likecoin-wallet-connector/pull/60/commits/5f187b73f617a9e44505fc1f1bfa60ecea3ae5ba)

**Review by nwingt — APPROVED** [review]

File: src/index.tsx [hunk 4]
```diff
@@ -359,16 +360,19 @@ export class LikeCoinWalletConnector {
...
+                <AuthcoreDialog
+                  onMount={({ containerId }) => {
+                    initAuthcore(this.options, { containerId });
+                  }}
+                  onClose={() => {
+                    this.closeDialog();
+                    resolve(undefined);
+                    this._events.emit('authcore_auth_closed');
+                  }}
+                  isHideSocialLogin={!!(this.options.authcoreClientId === 'likecoin-app-hidesocial')}
```
Comment (williamchong) [inline src/index.tsx:373]: Please name this `likecoin-app-hidesocial`  and `likecoin-app` as some constant.

Comment (AuroraHuang22) [reply src/index.tsx]: [🎨 Set the login social options as a constant](https://github.com/likecoin/likecoin-wallet-connector/pull/60/commits/8527d816b4ed2f3dd081028de36218dc10443689)

File: src/index.tsx [hunk 5]
```diff
@@ -64,6 +64,11 @@ const SESSION_KEY = 'likecoin_wallet_connector_session';
 
 const WC_BRIGDE = 'https://bridge.walletconnect.org';
 
+const SOCIAL_LOGIN_OPTIONS = {
+  HIDE_SOCIAL: 'likecoin-app-hidesocial',
+  DEFAULT: 'likecoin-app',
```
Comment (williamchong) [inline src/index.tsx:69]: Please also change
`authcoreClientId: options.authcoreClientId || 'likecoin-app',` in line 135

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/index.tsx (hunk 5 above)
Comment (AuroraHuang22) [reply src/index.tsx:69]: [🎨 Set the login social options as a constant](https://github.com/likecoin/likecoin-wallet-connector/pull/60/commits/40d1747944a3a8bc13671a833cfd7f64ee5be0db)

**Review by williamchong — APPROVED** [review]

**Pull request merged**